lms/djangoapps/course_api/blocks/views.py

```diff
@@ -0,0 +1,263 @@
+"""
+TODO
+"""
+from django.core.exceptions import ValidationError
+from django.http import Http404
+from rest_framework.generics import ListAPIView
+from rest_framework.response import Response
+
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import CourseKey
+from openedx.core.lib.api.view_utils import view_auth_classes, DeveloperErrorViewMixin
+from xmodule.modulestore.django import modulestore
+from xmodule.modulestore.exceptions import ItemNotFoundError
+
+from .api import get_blocks
+from .forms import BlockListGetForm
+
+
+@view_auth_classes()
+class BlocksView(DeveloperErrorViewMixin, ListAPIView):
+    """
+    **Use Case**
+
+        Returns the blocks within the requested block tree according to the requesting user's access level.
+
+    **Example requests**:
+
+        GET /api/courses/v1/blocks/<root_block_usage_id>/?depth=all
+        GET /api/courses/v1/blocks/<usage_id>/?
+            user=anjali,
+            &depth=all,
+            &requested_fields=graded,format,student_view_multi_device,
+            &block_counts=video,
+            &student_view_data=video,
+
+    **Parameters**:
+
+        * student_view_data: (list) Indicates for which block types to return student_view_data.
+
+          Example: student_view_data=video
+
+        * block_counts: (list) Indicates for which block types to return the aggregate count of the blocks.
+
+          Example: block_counts=video,problem
+
+        * requested_fields: (list) Indicates which additional fields to return for each block.
```

Comment:
I see now that it's in the response values. A note should be added here where to find the accepted values.